daemons/based/based_remote.c

```diff
@@ -445,11 +445,22 @@ cib_remote_msg(gpointer data)
     pcmk__client_t *client = ncd->new_client;
     xmlNode *command = NULL;
     int rc;
-    int timeout = 1000;
+    int timeout_sec = 1;
     const char *client_name = pcmk__client_name(client);
 
     if (pcmk_is_set(client->flags, pcmk__client_authenticated)) {
-        timeout = -1;
+        timeout_sec = 60;
+    }
+
+    /* If start_time is 0, we've previously handled a complete message and this
+     * callback is being reused for a new message.  Reset the start_time, giving
+     * this new message timeout_sec from now to complete.
+     */
+    if (ncd->start_time == 0) {
+        ncd->start_time = time(NULL);
+    } else if (time(NULL) >= ncd->start_time + timeout_sec) {
+        crm_info("Error reading from remote client: %s", pcmk_rc_str(ETIME));
+        return -1;
```

Comment:
It just occurred to me that we don't need a timeout on the server side. If a remote admin session is taking a long time to send a message, it doesn't bother the server in any way. It's not waiting for the message in order to do anything other than process that message. All that matters is the wait isn't blocking the server from doing other things.

The only possible issue I see is if a client is repeatedly opening connections and dropping them uncleanly, it might fill up the listen queue before the connections time out at the TCP level. But that seems like something to address with the (system-wide) TCP timeout.